spec/src/main/asciidoc/microprofile-lra-spec.adoc

```diff
@@ -387,6 +387,28 @@ will run in the context of an LRA and controls whether or not:
 - to throw an exception if the method returns particular HTTP status codes.
 - to enlist the class as a participant of the LRA.
 
+These operations are performed before the execution proceeds to the invoked
+`@LRA` annotated business method. If any of the LRA operations prohibits the
+execution of the business method, the LRA implemenation MUST return one of the
+following status codes to user invoking the LRA annotated method. This means
+that in these conditions the business method IS NOT invoked.
+
+|===
+|Status code |Conditions
+
+| 410
+a| - the passed LRA context has already ended
+   - the passed LRA context is not found
+
+| 412
+a| - the user passed the LRA context to LRA type which prohibits it (see LRA.Type)
+   - the content of the LRA context is invalid
```

Comment:
We do not mandate that implementations parse the context other than possibly validating that it conforms to [RFC 2396: Uniform Resource Identifiers (URI): Generic Syntax](https://www.ietf.org/rfc/rfc2396.txt). For example the context could have been created by another implementation and implementations should not need to understand the format used by other implementations. So if you are only thinking about URIs then its best that the text says so.

Reply:
-1, if the user passes `http://example.com` that is also a valid URI but I don't think any implementation will use it as an identifier. We explicitly don't define interoperability in 1.0 so I think that this text is correct.

Reply:
I do not understand your answer. Are you saying that if we only validate that the context conforms to RFC 2396 then we would end up allowing *http://example.com*?

Reply:
yes, this is what I understood that your answer is suggesting.

Reply:
The URI http://example.com conforms to RFC 2396 so it is valid. So my question is what does it mean for "the content of the LRA context" to be invalid?

Reply:
I would think yes, that is an invalid content and I would expect the implementation reply with 412 since it is a user error. Am I missing something?

Reply:
How can we know when a valid URI is not a valid LRAid? (from its structure, without checking if it is found internally which would be a 410 status if not found)

Reply:
+1

Reply:
ok, I will change it to 410